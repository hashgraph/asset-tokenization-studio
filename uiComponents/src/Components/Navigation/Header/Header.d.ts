import React from "react";
import type { FlexProps as ChakraFlexProps } from "@chakra-ui/react";
import type { BaseMultiStyleConfiguration } from "@/Theme/Components/BaseMultiStyleConfiguration";
export interface HeaderProps extends ChakraFlexProps {
    leftContent?: React.ReactElement;
    rightContent?: React.ReactElement;
    contentContainerProps?: ChakraFlexProps;
}
export declare const headerPartsList: Array<"container" | "contentContainer">;
type Parts = typeof headerPartsList;
export interface HeaderThemeConfiguration extends BaseMultiStyleConfiguration<Parts> {
}
export declare const Header: ({ leftContent, rightContent, contentContainerProps, ...props }: HeaderProps) => JSX.Element;
export {};
