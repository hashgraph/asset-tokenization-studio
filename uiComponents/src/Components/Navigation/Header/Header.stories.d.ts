declare const meta: import("@storybook/types").ComponentAnnotations<import("@storybook/react/dist/types-0a347bb9").R, import("./Header").HeaderProps>;
export default meta;
export declare const Default: import("@storybook/types").AnnotatedStoryFn<import("@storybook/react/dist/types-0a347bb9").R, import("./Header").HeaderProps>;
export declare const HeaderOnlyLeftContent: import("@storybook/types").AnnotatedStoryFn<import("@storybook/react/dist/types-0a347bb9").R, import("./Header").HeaderProps>;
export declare const HeaderOnlyRightContent: import("@storybook/types").AnnotatedStoryFn<import("@storybook/react/dist/types-0a347bb9").R, import("./Header").HeaderProps>;
