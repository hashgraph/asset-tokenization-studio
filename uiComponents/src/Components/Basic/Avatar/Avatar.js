var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
import { Avatar as ChakraAvatar, AvatarBadge as ChakraAvatarBadge, } from "@chakra-ui/avatar";
import { forwardRef, useMultiStyleConfig as useChakraMultiStyleConfig, } from "@chakra-ui/system";
import { SkeletonCircle as ChakraSkeletonCircle } from "@chakra-ui/react";
import { User } from "@phosphor-icons/react";
import React from "react";
import { PhosphorIcon, Weight } from "../../Foundations/PhosphorIcon";
export var avatarPartsList = [
    "container",
    "badge",
    "label",
    "group",
    "excessLabel",
];
export var Avatar = forwardRef(function (_a, ref) {
    var _b = _a.showBadge, showBadge = _b === void 0 ? false : _b, badgeColor = _a.badgeColor, size = _a.size, name = _a.name, src = _a.src, variant = _a.variant, isLoading = _a.isLoading, props = __rest(_a, ["showBadge", "badgeColor", "size", "name", "src", "variant", "isLoading"]);
    var styles = useChakraMultiStyleConfig("Avatar", {
        name: name && !src,
        size: size,
        badgeColor: badgeColor,
        variant: variant
    });
    if (isLoading) {
        return (React.createElement(ChakraSkeletonCircle, __assign({ sx: styles.container, size: size }, props)));
    }
    return (React.createElement(ChakraAvatar, __assign({ ref: ref, name: name, src: src, sx: styles.container, variant: variant, icon: React.createElement(PhosphorIcon, { weight: Weight.Regular, size: "sm", as: User }) }, props), showBadge && React.createElement(ChakraAvatarBadge, { sx: styles.badge })));
});
