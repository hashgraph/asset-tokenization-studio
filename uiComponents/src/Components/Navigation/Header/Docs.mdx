import {
  Story,
  Meta,
  Title,
  ArgTypes,
  Primary,
  Stories,
  Source,
} from "@storybook/addon-docs";
import { Grid, GridItem, Text, Link, Center, Flex } from "@chakra-ui/react";
import * as HeaderStories from "./Header.stories";

<Meta of={HeaderStories} />

<Title />

<a
  href="https://www.figma.com/file/YF8AvIq0MclfhGDvKgzypp/NEW---ioBricks-Library?node-id=1860-13948&t=3ozoVrWQYvgw1fj3-4"
  target="\_blank"
>
  View figma design
</a>

<Primary />

## How to set up

### Config theme

Add the following configurations to the `theme.tsx` and `Header.ts` files, respectively.

```tsx dark
const theme = {
  components: {
    Header: ConfigHeader,
  },
};
```

```tsx dark
const ConfigHeader: HeaderThemeConfiguration = {
  parts: headerPartsList,
  baseStyle: {},
};
```

### Component parts

You have these parts available: `"container" | "contentContainer"` (`headerPartsList`)

## Props

<ArgTypes of={HeaderStories} />

## Stories

<Stories title="" />
