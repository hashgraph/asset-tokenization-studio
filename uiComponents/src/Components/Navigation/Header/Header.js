var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
import React from "react";
import { useMultiStyleConfig as useChakraMultiStyleConfig } from "@chakra-ui/system";
import { Flex as ChakraFlex, Spacer as ChakraSpacer } from "@chakra-ui/react";
import { motion } from "framer-motion";
var MotionFlex = motion(ChakraFlex);
export var headerPartsList = [
    "container",
    "contentContainer",
];
export var Header = function (_a) {
    var leftContent = _a.leftContent, rightContent = _a.rightContent, contentContainerProps = _a.contentContainerProps, props = __rest(_a, ["leftContent", "rightContent", "contentContainerProps"]);
    var styles = useChakraMultiStyleConfig("Header");
    return (React.createElement(MotionFlex, __assign({ "data-testid": "header", sx: styles.container }, props),
        React.createElement(ChakraFlex, __assign({ sx: styles.contentContainer }, contentContainerProps),
            leftContent || React.createElement(ChakraSpacer, null),
            rightContent)));
};
