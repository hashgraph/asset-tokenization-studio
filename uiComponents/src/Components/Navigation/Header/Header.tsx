import React from "react";
import { useMultiStyleConfig as useChakraMultiStyleConfig } from "@chakra-ui/system";
import type { FlexProps as ChakraFlexProps } from "@chakra-ui/react";
import { Flex as ChakraFlex, Spacer as ChakraSpacer } from "@chakra-ui/react";
import type { BaseMultiStyleConfiguration } from "@/Theme/Components/BaseMultiStyleConfiguration";
import { motion } from "framer-motion";

const MotionFlex = motion(ChakraFlex);
export interface HeaderProps extends ChakraFlexProps {
  leftContent?: React.ReactElement;
  rightContent?: React.ReactElement;
  contentContainerProps?: ChakraFlexProps;
}

export const headerPartsList: Array<"container" | "contentContainer"> = [
  "container",
  "contentContainer",
];

type Parts = typeof headerPartsList;

export interface HeaderThemeConfiguration
  extends BaseMultiStyleConfiguration<Parts> {}

export const Header = ({
  leftContent,
  rightContent,
  contentContainerProps,
  ...props
}: HeaderProps) => {
  const styles = useChakraMultiStyleConfig("Header");

  return (
    <MotionFlex data-testid="header" sx={styles.container} {...props}>
      <ChakraFlex sx={styles.contentContainer} {...contentContainerProps}>
        {leftContent || <ChakraSpacer />}
        {rightContent}
      </ChakraFlex>
    </MotionFlex>
  );
};
