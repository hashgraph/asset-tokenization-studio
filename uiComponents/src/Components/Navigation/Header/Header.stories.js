var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
import React from "react";
import { Header } from "./Header";
import { Stack } from "@chakra-ui/react";
import { Logo } from "@/Components/Basic/Logo";
import { Text } from "@/Components/Foundations/Text";
import { Avatar } from "@/Components/Basic/Avatar";
var meta = {
    title: "Design System/Navigation/Header",
    component: Header,
    argTypes: {
        leftContent: { control: false, description: "Left content of the header" },
        rightContent: {
            control: false,
            description: "Right content of the header"
        },
        contentContainerProps: {
            control: false,
            description: "Used to modify the wrapper of the left and right content"
        }
    },
    parameters: {
        design: {
            type: "figma",
            url: "https://www.figma.com/file/YF8AvIq0MclfhGDvKgzypp/NEW---ioBricks-Library?node-id=1860-13948&t=3ozoVrWQYvgw1fj3-4"
        },
        docs: {}
    }
};
export default meta;
var Template = function (args) {
    return React.createElement(Header, __assign({}, args));
};
export var Default = Template.bind({});
Default.args = {
    leftContent: React.createElement(Logo, { alt: "IOB" }),
    rightContent: (React.createElement(Stack, { align: "center", direction: "row" },
        React.createElement(Text, { textStyle: "ElementsMediumXS" }, "Username"),
        React.createElement(Avatar, null)))
};
export var HeaderOnlyLeftContent = Template.bind({});
HeaderOnlyLeftContent.args = {
    leftContent: React.createElement(Logo, { alt: "IOB" })
};
export var HeaderOnlyRightContent = Template.bind({});
HeaderOnlyRightContent.args = {
    rightContent: (React.createElement(Stack, { align: "center", direction: "row" },
        React.createElement(Text, { textStyle: "ElementsMediumXS" }, "Username"),
        React.createElement(Avatar, null)))
};
