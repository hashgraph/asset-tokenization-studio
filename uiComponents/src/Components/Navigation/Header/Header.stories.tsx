import type { Meta, StoryFn } from "@storybook/react";
import React from "react";
import { Header } from "./Header";
import { Stack } from "@chakra-ui/react";
import { Logo } from "@/Components/Basic/Logo";
import { Text } from "@/Components/Foundations/Text";
import { Avatar } from "@/Components/Basic/Avatar";

const meta = {
  title: "Design System/Navigation/Header",
  component: Header,
  argTypes: {
    leftContent: { control: false, description: "Left content of the header" },
    rightContent: {
      control: false,
      description: "Right content of the header",
    },
    contentContainerProps: {
      control: false,
      description: "Used to modify the wrapper of the left and right content",
    },
  },
  parameters: {
    design: {
      type: "figma",
      url: "https://www.figma.com/file/YF8AvIq0MclfhGDvKgzypp/NEW---ioBricks-Library?node-id=1860-13948&t=3ozoVrWQYvgw1fj3-4",
    },
    docs: {},
  },
} as Meta<typeof Header>;
export default meta;

const Template: StoryFn<typeof Header> = (args) => {
  return <Header {...args} />;
};

export const Default = Template.bind({});
Default.args = {
  leftContent: <Logo alt="IOB" />,
  rightContent: (
    <Stack align="center" direction="row">
      <Text textStyle="ElementsMediumXS">Username</Text>
      <Avatar />
    </Stack>
  ),
};

export const HeaderOnlyLeftContent = Template.bind({});
HeaderOnlyLeftContent.args = {
  leftContent: <Logo alt="IOB" />,
};

export const HeaderOnlyRightContent = Template.bind({});
HeaderOnlyRightContent.args = {
  rightContent: (
    <Stack align="center" direction="row">
      <Text textStyle="ElementsMediumXS">Username</Text>
      <Avatar />
    </Stack>
  ),
};
