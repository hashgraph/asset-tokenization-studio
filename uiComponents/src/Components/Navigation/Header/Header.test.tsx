import { render } from "@/test-utils";
import React from "react";
import type { HeaderProps } from "./Header";
import { Header } from "./Header";
import { Stack } from "@chakra-ui/react";
import { Logo } from "@/Components/Basic/Logo";
import { Avatar } from "@/Components/Basic/Avatar";

describe(`<Header />`, () => {
  const defaultProps: HeaderProps = {
    leftContent: (
      <Stack data-testid="left-content" spacing={6}>
        <Logo alt="IOB" />
      </Stack>
    ),
    rightContent: (
      <div data-testid="right-content">
        <Avatar />
      </div>
    ),
  };
  const factoryComponent = (props: HeaderProps = defaultProps) =>
    render(<Header {...props} />);

  test("renders correctly", () => {
    const component = factoryComponent();
    expect(component.getByTestId("header")).toBeInTheDocument();
    expect(component.asFragment()).toMatchSnapshot();
  });
  test("renders correctly only left content", () => {
    const component = factoryComponent({
      leftContent: defaultProps.leftContent,
    });
    expect(component.getByTestId("left-content")).toBeVisible();
    expect(component.queryByTestId("right-content")).toBeNull();
    expect(component.asFragment()).toMatchSnapshot("OnlyLeftContent");
  });

  test("renders correctly only right content", () => {
    const component = factoryComponent({
      rightContent: defaultProps.rightContent,
    });
    expect(component.getByTestId("right-content")).toBeVisible();
    expect(component.queryByTestId("left-content")).toBeNull();
    expect(component.asFragment()).toMatchSnapshot("OnlyRightContent");
  });
});
