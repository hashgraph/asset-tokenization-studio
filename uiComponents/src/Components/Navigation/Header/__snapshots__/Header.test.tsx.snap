// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> renders correctly 1`] = `
<DocumentFragment>
  <div
    class="css-mo37bg"
    data-testid="header"
  >
    <div
      class="css-1k3povc"
    >
      <div
        class="chakra-stack css-b5leqj"
        data-testid="left-content"
      >
        <img
          alt="IOB"
          src="[object Object]"
          style="width: auto; height: auto;"
        />
      </div>
      <div
        data-testid="right-content"
      >
        <span
          class="chakra-avatar css-am7y12"
        >
          <svg
            aria-label=" avatar"
            class="chakra-icon css-e8vrb1"
            fill="currentColor"
            focusable="false"
            height="1em"
            role="img"
            viewBox="0 0 256 256"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M230.92,212c-15.23-26.33-38.7-45.21-66.09-54.16a72,72,0,1,0-73.66,0C63.78,166.78,40.31,185.66,25.08,212a8,8,0,1,0,13.85,8c18.84-32.56,52.14-52,89.07-52s70.23,19.44,89.07,52a8,8,0,1,0,13.85-8ZM72,96a56,56,0,1,1,56,56A56.06,56.06,0,0,1,72,96Z"
            />
          </svg>
        </span>
      </div>
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`<Header /> renders correctly only left content: OnlyLeftContent 1`] = `
<DocumentFragment>
  <div
    class="css-mo37bg"
    data-testid="header"
  >
    <div
      class="css-1k3povc"
    >
      <div
        class="chakra-stack css-b5leqj"
        data-testid="left-content"
      >
        <img
          alt="IOB"
          src="[object Object]"
          style="width: auto; height: auto;"
        />
      </div>
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`<Header /> renders correctly only right content: OnlyRightContent 1`] = `
<DocumentFragment>
  <div
    class="css-mo37bg"
    data-testid="header"
  >
    <div
      class="css-1k3povc"
    >
      <div
        class="css-17xejub"
      />
      <div
        data-testid="right-content"
      >
        <span
          class="chakra-avatar css-am7y12"
        >
          <svg
            aria-label=" avatar"
            class="chakra-icon css-e8vrb1"
            fill="currentColor"
            focusable="false"
            height="1em"
            role="img"
            viewBox="0 0 256 256"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M230.92,212c-15.23-26.33-38.7-45.21-66.09-54.16a72,72,0,1,0-73.66,0C63.78,166.78,40.31,185.66,25.08,212a8,8,0,1,0,13.85,8c18.84-32.56,52.14-52,89.07-52s70.23,19.44,89.07,52a8,8,0,1,0,13.85-8ZM72,96a56,56,0,1,1,56,56A56.06,56.06,0,0,1,72,96Z"
            />
          </svg>
        </span>
      </div>
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;
